vms/organization/views.py

```diff
@@ -2,105 +2,80 @@
 from django.core.urlresolvers import reverse
 from django.http import HttpResponse, HttpResponseRedirect
 from django.shortcuts import render
-
+from braces.views import LoginRequiredMixin, AnonymousRequiredMixin
+from django.views.generic.edit import FormView, UpdateView
+from django.views.generic.edit import DeleteView
+from django.views.generic import ListView
 from organization.forms import OrganizationForm
 from organization.services import *
+from organization.models import *
+from django.utils.decorators import method_decorator
+from django.core.urlresolvers import reverse_lazy
+
 
 
-@login_required
-def is_admin(request):
-    user = request.user
-    admin = None
-
-    try:
-        admin = user.administrator
-    except ObjectDoesNotExist:
-        pass
-
-    # check that an admin is logged in
-    if admin is not None:
-        return True
-    else:
-        return False
-
-
-@login_required
-def create(request):
-    if is_admin(request):
-        if request.method == 'POST':
-            form = OrganizationForm(request.POST)
-            if form.is_valid():
-                form.save()
-                return HttpResponseRedirect(reverse('organization:list'))
-            else:
-                return render(
-                    request,
-                    'organization/create.html',
-                    {'form': form, }
-                    )
+class AdministratorLoginRequiredMixin(object):
+
+    @method_decorator(login_required)
+    def dispatch(self, request, *args, **kwargs):
+        user = request.user
+        admin = None
+        try:
+            admin = user.administrator
+        except ObjectDoesNotExist:
+            pass
+        if not admin:
+            return render(request, 'vms/no_admin_rights.html')
         else:
-            form = OrganizationForm()
-            return render(
-                request,
-                'organization/create.html',
-                {'form': form, }
-                )
-    else:
-        return render(request, 'vms/no_admin_rights.html')
```

Comment:
@amruthasangeeth Where is template_name and form_class are redirecting to no_admin_rights in such a case?